site/src/components/DurationField/DurationField.tsx

```diff
@@ -0,0 +1,115 @@
+import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+import MenuItem from "@mui/material/MenuItem";
+import Select from "@mui/material/Select";
+import TextField from "@mui/material/TextField";
+import { useState, type FC } from "react";
+
+type TimeUnit = "days" | "hours";
+
+// Value should be in milliseconds or undefined. Undefined means no value.
+type DurationValue = number | undefined;
+
+type DurationFieldProps = {
+  label: string;
+  value: DurationValue;
+  onChange: (value: DurationValue) => void;
+};
+
+export const DurationField: FC<DurationFieldProps> = (props) => {
+  const { label, value, onChange } = props;
+  const [timeUnit, setTimeUnit] = useState<TimeUnit>(() => {
+    if (!value) {
+      return "hours";
+    }
+
+    return Number.isInteger(durationToDays(value)) ? "days" : "hours";
+  });
+
+  return (
+    <div
+      css={{
+        display: "flex",
+        gap: 8,
+      }}
+    >
+      <TextField
+        css={{ maxWidth: 160 }}
+        label={label}
+        value={
+          !value
+            ? ""
+            : timeUnit === "hours"
+              ? durationToHours(value)
+              : durationToDays(value)
+        }
+        onChange={(e) => {
+          if (e.target.value === "") {
+            onChange(undefined);
+          }
+
+          const value = parseInt(e.target.value);
+
+          if (Number.isNaN(value)) {
+            return;
+          }
+
+          onChange(
+            timeUnit === "hours"
+              ? hoursToDuration(value)
+              : daysToDuration(value),
+          );
+        }}
```

Comment:
Could we swap in `e.target.value` with `e.target.valueAsNumber`? We'd still need to catch `NaN` values, but I think it'd simplify the logic a little more

Reply:
I always forget about `valueAsNumber `

Reply:
Looks like it is not a valid API on our current TS version.

<img width="689" alt="Screenshot 2024-05-08 at 16 17 23" src="https://github.com/coder/coder/assets/3165839/65ddabb0-17a8-49d9-8759-732517fefbea">

Or maybe the TextField is handling types differently because it has to support textarea as well.